src/TemplateDialog.js

```diff
@@ -0,0 +1,203 @@
+import React, { PropTypes } from 'react'
+import { connect } from 'react-redux'
+import $ from 'jquery'
+
+import LabeledSelect from './LabeledSelect'
+import LabeledTextField from './LabeledTextField'
+import DetailContainer from './DetailContainer'
+import ErrorAlert from './ErrorAlert'
+import Selectors from './selectors'
+import {
+  updateEditTemplateName,
+  updateEditTemplateDescription,
+  updateEditTemplateOS,
+  updateEditTemplateMemory,
+  updateEditTemplateCpu,
+  changeEditTemplate,
+  editTemplate,
+  closeDetail,
+} from './actions'
+
+class TemplateDialog extends React.Component {
+  constructor (props) {
+    super(props)
+    this.changeTemplate = this.changeTemplate.bind(this)
+    this.changeOperatingSystem = this.changeOperatingSystem.bind(this)
+    this.changeMemory = this.changeMemory.bind(this)
+    this.changeCpu = this.changeCpu.bind(this)
+    this.changeName = this.changeName.bind(this)
+    this.changeDescription = this.changeDescription.bind(this)
+    this.editTemplate = this.editTemplate.bind(this)
+    this.closeDialog = this.closeDialog.bind(this)
+  }
+
+  componentDidUpdate () {
+    // without this template combobox its not rerendering
+    $(this.templateOs).selectpicker('refresh')
+  }
+
+  componentDidMount () {
+    $(this.template).selectpicker()
+    $(this.templateOs).selectpicker()
+  }
+
+  changeName () {
+    this.props.changeName(this.templateName.value)
+  }
+
+  changeTemplate () {
+    this.props.changeTemplate(Selectors.getTemplateByName(this.template.value))
+  }
+
+  changeOperatingSystem () {
+    this.props.changeOperatingSystem(this.templateOs.value)
+  }
+
+  changeDescription () {
+    this.props.changeDescription(this.templateDescription.value)
+  }
+
+  changeMemory () {
+    this.props.changeMemory(this.templateMemory.value)
+  }
+
+  changeCpu () {
+    this.props.changeCpu(this.templateCpus.value)
+  }
+
+  closeDialog (e) {
+    e.preventDefault()
+    this.props.closeDialog()
+  }
+
+  editTemplate (e) {
+    e.preventDefault()
+    const template = {
+      'name': this.templateName.value,
+      'description': this.templateDescription.value,
+      'memory': this.templateMemory.value,
+      'cpu': {
+        'topology': {
+          'cores': '1',
+          'sockets': this.templateCpus.value,
+          'threads': '1',
+        },
+      },
+      'os': {
+        'type': this.templateOs.value,
+      },
+    }
+    this.props.editTemplate(template, this.props.template.get('id'))
+  }
+
+  render () {
+    return (
+      <DetailContainer>
+        <h1>Edit Template</h1>
```

Comment:
This whole thing is from large part a copy/paste of the VmDialog. There are couple of issues with this, namely:
- when this becomes bigger, it becomes a maintenance nightmare
- when this will edit more entities (pools, instance types) it will become a maintenance nightmare on steroids (trust me, we had a similar issue with this dialog on the ovirt and we were drowning in in bugs like "this field is working in edit VM and Pool, but does not work in instance type" etc.)
- even now when it is small you have a bug like this here :) The memory label in VM has the changed label "Memory (MB)" and here it is still "Memory".